mmv1/products/clouddomains/Registration.yaml

```diff
@@ -0,0 +1,522 @@
+        name: currencyCode
+        description: The three-letter currency code defined in ISO 4217.
+      - !ruby/object:Api::Type::String
+        name: units
+        description: The whole units of the amount. For example if currencyCode is "USD", then 1 unit is one US dollar.
+  - !ruby/object:Api::Type::Array
+    name: supportedPrivacy
+    description: Output only. Set of options for the contactSettings.privacy field that this Registration supports.
+    item_type: Api::Type::String
+    output: true
+  - !ruby/object:Api::Type::String
+    name: registerFailureReason
+    output: true
+    description: Output only. The reason the domain registration failed. Only set for domains in REGISTRATION_FAILED state.
+  - !ruby/object:Api::Type::NestedObject
+    name: managementSettings
+    description: "Settings for management of the Registration, including renewal, billing, and transfer"
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: renewalMethod
+        output: true
+        description: |
+          Output only. The actual renewal method for this Registration. When preferredRenewalMethod is set to AUTOMATIC_RENEWAL,
+          the actual renewalMethod can be equal to RENEWAL_DISABLED—for example, when there are problems with the billing account
+          or reported domain abuse. In such cases, check the issues field on the Registration. After the problem is resolved, the
+          renewalMethod is automatically updated to preferredRenewalMethod in a few hours.
+      - !ruby/object:Api::Type::String
+        name: preferredRenewalMethod
+        default_from_api: true
+        description: |
+          The desired renewal method for this Registration. The actual renewalMethod is automatically updated to reflect this choice.
+          If unset or equal to RENEWAL_METHOD_UNSPECIFIED, the actual renewalMethod is treated as if it were set to AUTOMATIC_RENEWAL.
+          You cannot use RENEWAL_DISABLED during resource creation, and you can update the renewal status only when the Registration
+          resource has state ACTIVE or SUSPENDED.
+
+          When preferredRenewalMethod is set to AUTOMATIC_RENEWAL, the actual renewalMethod can be set to RENEWAL_DISABLED in case of
+          problems with the billing account or reported domain abuse. In such cases, check the issues field on the Registration. After
+          the problem is resolved, the renewalMethod is automatically updated to preferredRenewalMethod in a few hours.
+      - !ruby/object:Api::Type::String
+        name: transferLockState
+        default_from_api: true
+        description: Controls whether the domain can be transferred to another registrar. Values are UNLOCKED or LOCKED.
+  - !ruby/object:Api::Type::NestedObject
+    name: dnsSettings
+    description: Settings controlling the DNS configuration of the Registration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: customDns
```

Comment:
conflict with `googleDomainsDns`? since it's union field `dns_provider` according to the [API doc](https://cloud.google.com/domains/docs/reference/rest/v1beta1/projects.locations.registrations#dnssettings)

Reply:
Good point, I'll add this

Reply:
Is the union field `dns_provider` required? as I see we're adding `exactly_one_of` here